osmoutils/cosmwasm/helpers.go

```diff
@@ -51,6 +54,7 @@ func Query[T any, K any](ctx sdk.Context, wasmKeeper WasmKeeper, contractAddress
 		return response, err
 	}
 
+	ctx = ctx.WithGasMeter(storetypes.NewGasMeter(wasmKeeper.QueryGasLimit()))
```

Comment:
If this is in consensus, then this is state breaking. I think protorev handled this, let me see if I can find a link

Reply:
This is only used in cosmwasmpool for CalcInAmountGivenOut (and the other dir)

can someone sanity check those are only used in queries? If so looks good to merge imo

Reply:
Im pretty sure this is called within consensus :/

SwapExactAmountOut -> RouteExactAmountOut -> createMultihopExpectedSwapOuts -> swapModule.CalcInAmtGivenOut